activation/activation.go

```diff
@@ -775,3 +779,7 @@ func SignAtx(signer signer, atx *types.ActivationTx) error {
 	atx.Sig = signer.Sign(bts)
 	return nil
 }
+
+func isStale(challenge *types.NIPostChallenge, currentEpoch types.EpochID) bool {
+	return challenge.TargetEpoch() < currentEpoch
```

Comment:
Is this correct? Lets say `PublishEpoch` is , `TargetEpoch` is 5 and `currentEpoch` is also 5 can the ATX still be published? I think this should be:

```suggestion
	return challenge.PublishEpoch() < currentEpoch
```

wdyt?

Reply:
Yes, it is OK (not ideal but technically still correct) to publish an ATX in the target epoch. Please see the [discussion](https://spacemesh.slack.com/archives/C89QJJY3G/p1672396924995409) (sorry for private link). Copying over the most important part:

> target epoch is when atx is actually used. there are multiple downsides if someone will do that
for smesher the downside is that it already missed some amount of rewards
for the network - average number of proposals/ballots will be higher than expected (though i am not fully confident if everything is correct here)
i think the codebase should optimize for not doing this. but if smesher finds itself in such condition it probably should still publish it, otherwise it won't get any rewards for the whole epoch